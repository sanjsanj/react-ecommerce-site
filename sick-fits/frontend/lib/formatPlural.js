exports.formatPlural = (value, string) =>
  string + (value.length === 1 ? "" : string.slice(-1) === "s" ? "es" : "s");
