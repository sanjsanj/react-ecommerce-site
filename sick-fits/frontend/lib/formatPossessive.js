exports.formatPossessive = word =>
  `${word}'${word.slice(-1) === "s" ? "" : "s"}`;
