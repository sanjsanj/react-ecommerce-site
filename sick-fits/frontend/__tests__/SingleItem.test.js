import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import waait from "waait";
import { MockedProvider } from "react-apollo/test-utils";

import { fakeItem } from "../lib/testUtils";

import SingleItem, { SINGLE_ITEM_QUERY } from "../components/SingleItem";

describe("<SingleItem />", () => {
  it("renders", async () => {
    const mocks = [
      {
        request: {
          query: SINGLE_ITEM_QUERY,
          variables: {
            id: "123"
          }
        },
        result: {
          data: {
            item: fakeItem()
          }
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );

    expect(wrapper.text()).toContain("Loading...");

    await waait();
    wrapper.update();

    expect(toJSON(wrapper.find("h2"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("img"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("p"))).toMatchSnapshot();
  });

  it("errors when item not found", async () => {
    const mocks = [
      {
        request: {
          query: SINGLE_ITEM_QUERY,
          variables: {
            id: "123"
          }
        },
        result: {
          errors: [
            {
              message: "Item not found"
            }
          ]
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );

    await waait();
    wrapper.update();

    const item = wrapper.find("[data-test='graphql-error']");
    
    expect(item.text()).toContain("Item not found");

    expect(toJSON(item)).toMatchSnapshot();
  });
});
