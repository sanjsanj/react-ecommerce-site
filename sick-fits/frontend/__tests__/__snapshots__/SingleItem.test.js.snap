// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> errors when item not found 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Shoot!
  </strong>
  Item not found
</p>
`;

exports[`<SingleItem /> renders 1`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;

exports[`<SingleItem /> renders 2`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;

exports[`<SingleItem /> renders 3`] = `
<p>
  dogs
</p>
`;
